server/accounts_test.go

```diff
@@ -0,0 +1,835 @@
+		}
+	}
+
+	// Now check for the imports and exports of streams and services.
+	if lis := len(natsAcc.imports.streams); lis != 2 {
+		t.Fatalf("Expected 2 imported streams, got %d\n", lis)
+	}
+	if lis := len(natsAcc.imports.services); lis != 1 {
+		t.Fatalf("Expected 1 imported service, got %d\n", lis)
+	}
+	if les := len(natsAcc.exports.services); les != 1 {
+		t.Fatalf("Expected 1 exported service, got %d\n", les)
+	}
+	if les := len(natsAcc.exports.streams); les != 0 {
+		t.Fatalf("Expected no exported streams, got %d\n", les)
+	}
+
+	if synAcc == nil {
+		t.Fatalf("Error retrieving account for 'synadia'")
+	}
+
+	if lis := len(synAcc.imports.streams); lis != 0 {
+		t.Fatalf("Expected no imported streams, got %d\n", lis)
+	}
+	if lis := len(synAcc.imports.services); lis != 1 {
+		t.Fatalf("Expected 1 imported service, got %d\n", lis)
+	}
+	if les := len(synAcc.exports.services); les != 2 {
+		t.Fatalf("Expected 2 exported service, got %d\n", les)
+	}
+	if les := len(synAcc.exports.streams); les != 2 {
+		t.Fatalf("Expected 2 exported streams, got %d\n", les)
+	}
+}
+
+func TestImportExportConfigFailures(t *testing.T) {
+	// Import from unknow account
+	cf := createConfFile(t, []byte(`
+    accounts {
+      nats.io {
+        imports = [{stream: {account: "synadia", subject:"foo"}}]
+      }
+    }
+    `))
+	defer os.Remove(cf)
+	if _, err := ProcessConfigFile(cf); err == nil {
+		t.Fatalf("Expected an error with import from unknown account")
+	}
+	// Import a service with no account.
+	cf = createConfFile(t, []byte(`
+    accounts {
+      nats.io {
+        imports = [{service: subject:"foo.*"}]
+      }
+    }
+    `))
+	defer os.Remove(cf)
+	if _, err := ProcessConfigFile(cf); err == nil {
+		t.Fatalf("Expected an error with import of a service with no account")
+	}
+	// Import a service with a wildcard subject.
+	cf = createConfFile(t, []byte(`
+    accounts {
+      nats.io {
+        imports = [{service: {account: "nats.io", subject:"foo.*"}]
+      }
+    }
+    `))
+	defer os.Remove(cf)
+	if _, err := ProcessConfigFile(cf); err == nil {
+		t.Fatalf("Expected an error with import of a service with wildcard subject")
+	}
+	// Export with unknown keyword.
+	cf = createConfFile(t, []byte(`
+    accounts {
+      nats.io {
+        exports = [{service: "foo.*", wat:true}]
+      }
+    }
+    `))
+	defer os.Remove(cf)
+	if _, err := ProcessConfigFile(cf); err == nil {
+		t.Fatalf("Expected an error with export with unknown keyword")
+	}
+	// Import with unknown keyword.
+	cf = createConfFile(t, []byte(`
+    accounts {
+      nats.io {
+        imports = [{stream: {account: nats.io, subject: "foo.*"}, wat:true}]
+      }
+    }
+    `))
+	defer os.Remove(cf)
+	if _, err := ProcessConfigFile(cf); err == nil {
+		t.Fatalf("Expected an error with import with unknown keyword")
+	}
+	// Export with an account.
+	cf = createConfFile(t, []byte(`
+    accounts {
+      nats.io {
+        exports = [{service: {account: nats.io, subject:"foo.*"}]
```

Comment:
There is a missing bracket in this one (so right now the error is a parsing error)

```diff
diff --git a/server/accounts_test.go b/server/accounts_test.go
index eee29b3..dda4a25 100644
--- a/server/accounts_test.go
+++ b/server/accounts_test.go
@@ -423,13 +423,15 @@ func TestImportExportConfigFailures(t *testing.T) {
 	cf = createConfFile(t, []byte(`
     accounts {
       nats.io {
-        exports = [{service: {account: nats.io, subject:"foo.*"}]
+        exports = [{service: {account: nats.io, subject:"foo.*"}}]
       }
     }
     `))
```

Reply:
fixed